java/cdk-custom-resources/src/test/java/sleeper/cdk/custom/TableDefinerLambdaIT.java

```diff
@@ -114,7 +149,212 @@ public void shouldCreateTableWithMultipleSplitPoints() throws IOException {
                         .buildList());
     }
 
+    @Test
+    public void shouldFailToDeleteTableThatDoesNotExist() {
+        // Given
+        HashMap<String, Object> resourceProperties = new HashMap<>();
+        resourceProperties.put("tableProperties", tableProperties.saveAsString());
+
+        CloudFormationCustomResourceEvent event = CloudFormationCustomResourceEvent.builder()
+                .withRequestType("Delete")
+                .withResourceProperties(resourceProperties)
+                .build();
+
+        // When / Then
+        assertThatThrownBy(() -> tableDefinerLambda.handleEvent(event, null))
+                .isInstanceOf(TableNotFoundException.class);
+
+    }
+
+    @Test
+    void shouldTakeTableOfflineWhenDeleteCalledAndInstancePropertySetFalse() throws Exception {
+        // Given
+        instanceProperties.set(RETAIN_DATA_AFTER_TABLE_REMOVAL, "true");
+        S3InstanceProperties.saveToS3(s3Client, instanceProperties);
+
+        String tableName = tableProperties.get(TABLE_NAME);
+        TableProperties tableProps = createTable(uniqueIdAndName(tableProperties.get(TABLE_ID), tableName));
+        StateStore stateStore = createStateStore(tableProps);
+        update(stateStore).initialise(schema);
+        AllReferencesToAFile file = ingestRows(tableProps, List.of(new Row(Map.of("key1", 25L))));
+
+        // When
+        HashMap<String, Object> resourceProperties = new HashMap<>();
+        resourceProperties.put("tableProperties", tableProperties.saveAsString());
+        CloudFormationCustomResourceEvent event = CloudFormationCustomResourceEvent.builder()
+                .withRequestType("Delete")
+                .withResourceProperties(resourceProperties)
+                .build();
+        tableDefinerLambda.handleEvent(event, null);
+
+        // Then
+        tableProps.set(TABLE_ONLINE, "false");
+        assertThat(propertiesStore.loadByName(tableName)).isEqualTo(tableProps);
+        assertThat(streamTableFileTransactions(tableProps)).isNotEmpty();
+        assertThat(streamTablePartitionTransactions(tableProps)).isNotEmpty();
+        assertThat(listDataBucketObjectKeys())
+                .extracting(FilenameUtils::getName)
+                .containsExactly(
+                        FilenameUtils.getName(file.getFilename()),
+                        FilenameUtils.getName(file.getFilename()).replace("parquet", "sketches"));
+    }
+
+    @Test
+    void shouldFullyDeleteSpecifiedTableButNoOthers() throws Exception {
+        // Given
+        instanceProperties.set(RETAIN_DATA_AFTER_TABLE_REMOVAL, "false");
+        S3InstanceProperties.saveToS3(s3Client, instanceProperties);
+
+        String tableName = tableProperties.get(TABLE_NAME);
+        TableProperties table1 = createTable(uniqueIdAndName(tableProperties.get(TABLE_ID), tableName));
```

Comment:
It doesn't look like there's a reason to create a separate table properties object here if you're copying values from the original one?

You could either remove the createTable method, or not use the tableProperties field at all for this test?